air/src/air/mod.rs

```diff
@@ -285,6 +291,24 @@ pub trait Air: Send + Sync {
     // PROVIDED METHODS
     // --------------------------------------------------------------------------------------------
 
+    /// Returns the random elements to use in the lagrange kernel.
+    ///
+    /// The return slice stores the least significant random element first. For example, for a trace
+    /// length of 8, there will be 3 random elements, such that
+    /// - return[0] = r0
+    /// - return[1] = r1
+    /// - return[2] = r2
+    /// where `return` is the returned slice, and r = (r2, r1, r0).
+    fn lagrange_kernel_rand_elements<'r, F, E>(&self, aux_rand_elements: &'r [E]) -> &'r [E]
```

Comment:
I'm not sure we need this function just yet. I think for now we should assume that $r$-values are exogenous to the system. In the next PR, we can figure out where these should come from.

Reply:
removed